tests/FunctionalJavascript/AddCorporateFormTest.php

```diff
@@ -0,0 +1,200 @@
+
+  /**
+   * An test user with permission to create contact forms.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $testuser;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'user',
+    'oe_contact_forms',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create and login test user.
+    $this->testuser = $this->drupalCreateUser([
+      'administer contact forms',
+    ]);
+    $this->drupalLogin($this->testuser);
+  }
+
+  /**
+   * Tests the corporate field requirements.
+   *
+   * First we test requirements for state dependent fields,
+   * then we test the ajax add and remove,
+   * finally we submit and see if values are saved.
+   */
+  public function testAddCorporateForm(): void {
+    $this->drupalGet('admin/structure/contact/add');
+
+    /** @var \Behat\Mink\WebAssert $assert */
+    $assert = $this->assertSession();
+    /** @var \Behat\Mink\Element\DocumentElement $page */
+    $page = $this->getSession()->getPage();
+
+    // Assert new form fields are hidden at first.
+    $topic_name = $page->findField('topics_fieldset[group][0][topic_name]');
+    $this->assertNotEmpty($topic_name);
+    $this->assertFalse($topic_name->isVisible());
+    $topic_email_address = $page->findField('topics_fieldset[group][0][topic_email_address]');
+    $this->assertNotEmpty($topic_email_address);
+    $this->assertFalse($topic_email_address->isVisible());
+    $topic_label = $page->findField('topic_label');
+    $this->assertNotEmpty($topic_label);
+    $this->assertFalse($topic_label->isVisible());
+    $email_subject = $page->findField('email_subject');
+    $this->assertNotEmpty($email_subject);
+    $this->assertFalse($email_subject->isVisible());
+    $header = $page->findField('header');
+    $this->assertNotEmpty($header);
+    $this->assertFalse($header->isVisible());
+    $privacy_policy = $page->findField('privacy_policy');
+    $this->assertNotEmpty($privacy_policy);
+    $this->assertFalse($privacy_policy->isVisible());
+    $includes_fields_in_auto_reply = $page->findField('includes_fields_in_auto_reply');
+    $this->assertNotEmpty($includes_fields_in_auto_reply);
+    $this->assertFalse($includes_fields_in_auto_reply->isVisible());
+    $allow_canonical_url = $page->findField('allow_canonical_url');
+    $this->assertNotEmpty($allow_canonical_url);
+    $this->assertFalse($allow_canonical_url->isVisible());
+    $expose_as_block = $page->findField('expose_as_block');
+    $this->assertNotEmpty($expose_as_block);
+    $this->assertFalse($expose_as_block->isVisible());
+    $oe_country_residence = $page->findField('optional_fields[oe_country_residence]');
+    $this->assertNotEmpty($oe_country_residence);
+    $this->assertFalse($oe_country_residence->isVisible());
+    $oe_telephone = $page->findField('optional_fields[oe_telephone]');
+    $this->assertNotEmpty($oe_telephone);
+    $this->assertFalse($oe_telephone->isVisible());
```

Comment:
May be we can create an array of fields to check and loop throw it

foreach ([
  'topic_name' => 'topics_fieldset[group][0][topic_name]',
] as $variable_name => $field_path){
  $$variable_name = $page->findField($field_path);
  $this->assertNotEmpty($$variable_name);
  $this->assertFalse($$variable_name->isVisible());
}

Reply:
done

Reply:
I love you commit.